Tasks/Common/artifacts-common/connectionDataUtils.ts

```diff
@@ -29,6 +29,35 @@ export async function getConnectionDataForProtocol(protocolType: protocols.Proto
     }, 3, 1000);
 }
 
+/**
+ * Gets the api location url for any feed given the location id.
+ * @param protocolType Packaging protoocol like "NuGet", "PyPI"
+ * @param apiVersion Api version of the endpoint. ex: 3.0-preview, 5.0
+ * @param locationGuid location id for an api. ex: 93377A2C-F5FB-48B9-A8DC-7781441CABF1 for PyPi simple api
+ * @param feedId FeedId to put in api params
+ * @param project projectId to put in api params
+ */
+export async function getPackagingEndpointUrl(
+    protocolType: protocols.ProtocolType,
+    apiVersion: string,
+    locationGuid: string,
+    feedId: string,
+    project: string): Promise<string> {
+
+    tl.debug('Finding the URI for the packaging service');
+    return await retryOnException(async () => {
+        const accessToken = api.getSystemAccessToken();
+        const areaId = protocols.getAreaIdForProtocol(protocolType);
+        const serviceUri = await getServiceUriFromAreaId(areaId, accessToken);
+
+        const webApi = api.getWebApiWithProxy(serviceUri, accessToken);
+        const data = await webApi.vsoClient.getVersioningData(apiVersion, protocols.ProtocolType[protocolType], locationGuid, { feedId: feedId, project: project });
+
+        tl.debug('Feed registry url: ' + data.requestUrl);
```

Comment:
"Feed registry url" doesn't really accurately describe what this returns. Perhaps something like "Resolved route: "